demo/src/App.js

```diff
@@ -33,10 +54,27 @@ export function App() {
                 // eslint-disable-next-line max-len
                 d="M0 1.5A.5.5 0 0 1 .5 1H2a.5.5 0 0 1 .485.379L2.89 3H14.5a.5.5 0 0 1 .491.592l-1.5 8A.5.5 0 0 1 13 12H4a.5.5 0 0 1-.491-.408L2.01 3.607 1.61 2H.5a.5.5 0 0 1-.5-.5zM3.102 4l1.313 7h8.17l1.313-7H3.102zM5 12a2 2 0 1 0 0 4 2 2 0 0 0 0-4zm7 0a2 2 0 1 0 0 4 2 2 0 0 0 0-4zm-7 1a1 1 0 1 0 0 2 1 1 0 0 0 0-2zm7 0a1 1 0 1 0 0 2 1 1 0 0 0 0-2z"/>
             </svg>
+            <Badge
+              className="shopping-cart-badge"
+              variant={itemsInCart > 0 ? 'danger' : 'secondary'}
+              pill
+            >
+              {itemsInCart}
+            </Badge>
           </Nav.Link>
         </Nav>
       </Navbar>
       <Router/>
     </div>
   );
-}
+};
+
+AppBase.propTypes = {
+  items: PropTypes.object,
+};
+
+const mapStateToProps = (state) => ({
+  items: state.items,
+});
```

Comment:
Also for my understanding: this + the connect below is saying that rather than reading items from the HTML attribute items, it should come from the app's state.items JS property?

Reply:
Yes this is to get the items from the state